To: vim_dev@googlegroups.com
Subject: Patch 7.3.151
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.151 (after 7.3.074)
Problem:    When "unnamedplus" is in 'clipboard' the selection is sometimes
	    also copied to the star register.
Solution:   Avoid copy to the star register when undesired. (James Vega)
Files:	    src/ops.c


*** ../vim-7.3.150/src/ops.c	2010-12-08 14:23:08.000000000 +0100
--- src/ops.c	2011-04-01 16:23:10.000000000 +0200
***************
*** 3148,3157 ****
  	    /* Copy the text from register 0 to the clipboard register. */
  	    copy_yank_reg(&(y_regs[PLUS_REGISTER]));
  
- 	/* No need to copy to * register upon 'unnamed' now - see below */
  	clip_own_selection(&clip_plus);
  	clip_gen_set_selection(&clip_plus);
! 	if (!clip_isautosel() && !did_star)
  	{
  	    copy_yank_reg(&(y_regs[STAR_REGISTER]));
  	    clip_own_selection(&clip_star);
--- 3148,3156 ----
  	    /* Copy the text from register 0 to the clipboard register. */
  	    copy_yank_reg(&(y_regs[PLUS_REGISTER]));
  
  	clip_own_selection(&clip_plus);
  	clip_gen_set_selection(&clip_plus);
! 	if (!clip_isautosel() && !did_star && curr == &(y_regs[PLUS_REGISTER]))
  	{
  	    copy_yank_reg(&(y_regs[STAR_REGISTER]));
  	    clip_own_selection(&clip_star);
*** ../vim-7.3.150/src/version.c	2011-04-01 16:07:41.000000000 +0200
--- src/version.c	2011-04-01 16:25:40.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     151,
  /**/

-- 
BLACK KNIGHT: I'm invincible!
ARTHUR:       You're a looney.
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
